### PR TITLE
[cda] Try to fix and improve extraction (fixes #24458)

### LINKED ISSUE #24458: cda.pl Broken Extractor
## Checklist

- [X] I'm reporting a broken site support
- [X] I've verified that I'm running youtube-dl version **2020.03.24**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [X] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://www.cda.pl/video/14954728b', u'--verbose']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.03.24
[debug] Python version 2.7.5 (CPython) - Linux-3.10.0-1062.12.1.el7.x86_64-x86_64-with-centos-7.7.1908-Core
[debug] exe versions: none
[debug] Proxy map: {}
[CDA] 14954728b: Downloading webpage
[CDA] 14954728b: Downloading 360p version information
[CDA] 14954728b: Downloading 720p version information
[debug] Default format spec: best/bestvideo+bestaudio
[debug] Invoking downloader on u'G86C_bg%5D452%5DA%3D%5EE8%3A%23Hp08hAy8gtczBCurF8%5E%60dgd_hffec%5ED557a6_afae4%605g572gggbc56gdd2e3%6024'
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 474, in main
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 464, in _real_main
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2019, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 808, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 863, in process_ie_result
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1644, in process_video_result
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1926, in process_info
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1865, in dl
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/common.py", line 366, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/http.py", line 341, in real_download
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/http.py", line 109, in establish_connection
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2238, in urlopen
  File "/usr/lib64/python2.7/urllib2.py", line 423, in open
    protocol = req.get_type()
  File "/usr/lib64/python2.7/urllib2.py", line 285, in get_type
    raise ValueError, "unknown url type: %s" % self.__original
ValueError: unknown url type: G86C_bg%5D452%5DA%3D%5EE8%3A%23Hp08hAy8gtczBCurF8%5E%60dgd_hffec%5ED557a6_afae4%605g572gggbc56gdd2e3%6024

```

## Description

My guess is that they have encrypted displayed urls to base64 but i can be wrong.

### PR DESCRIPTION
**Pull request by divadsn**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixed url extraction from cda.pl links after a recent player.js update on the website made it harder to decode the file from `player_data`. I took the player.js code apart and found the method responsible for decrypting the file url and implemented the new method into the extractor.

### CODE REVIEW

**Comment by divadsn** [comment]
Do we still have to support Python 2 despite of it's EOL since January 2020? :thinking:

**Comment by vane** [comment]
Checked out your remote, thanks. It works for me.

**Comment by divadsn** [comment]
Since the new CDA.pl player code still contains the old video file decryption method, I decided to keep both methods to ensure best compatibility with all links that have not been tested.

**Comment by divadsn** [comment]
@dstftw could you review please?

**Comment by PiotrDabrowskey** [comment]
Can also confirm this fixes the problem. Thanks!

**Comment by Bartixxx32** [comment]
Thanks :D please review @dstftw

File: youtube_dl/extractor/cda.py [hunk 1]
```diff
@@ -137,7 +160,9 @@ def extract_format(page, version):
             if not video or 'file' not in video:
                 self.report_warning('Unable to extract %s version information' % version)
                 return
-            if video['file'].startswith('uggc'):
+            if "http" not in video['file'] and ".mp4" not in video['file'] and "uggcf://" not in video['file']:
```
Comment (dstftw) [inline youtube_dl/extractor/cda.py:163]: Inconsistent quotes.

**Pull request closed without merging**